metadata/src/main/java/org/apache/kafka/controller/QuorumController.java

```diff
@@ -858,7 +858,7 @@ private void rescheduleMaybeFenceStaleBrokers() {
             return;
         }
         scheduleDeferredWriteEvent(MAYBE_FENCE_REPLICAS, nextCheckTimeNs, () -> {
-            ControllerResult<Void> result = replicationControl.maybeFenceStaleBrokers();
+            ControllerResult<Void> result = replicationControl.maybeFenceOneStaleBroker();
             rescheduleMaybeFenceStaleBrokers();
```

Comment:
One thing I was considering is whether we should schedule the next check right away. For example, maybe we could check whether `result` contains a non-empty record set and reschedule immediately (or perhaps after a short backoff) if it does.

Reply:
I thought about doing that, but then realized that a call to reschedule will automatically result in a scheduling of another invocation right away if there were any brokers that were deemed stale still remaining in the unfenced list.

This will happen because the `nextCheckTimeNs` is based on when the broker with the older `lastContactNs` time (broker at the head of the list) would time out. This will always result in a time less than now if we have a stale broker and there should be no delay in executing this again (barring queue depth).

I also did not want to "prioritize" (if that's even possible) over other requests, because my understanding is that the fix works as long as a subsequent fence broker call is made after the first has been processed by the controller thread (which would also happen by picking up the record from the queue and replaying it).

Please let me know if my understanding is off here.

Reply:
It sounds like you are saying there may still be a race condition if the next expiration is scheduled too quickly. However, I think the way the code works, we are guaranteed to have this result reflected before the next scheduled operation can trigger. It would be good to have a test covering this scenario.

Reply:
I agree that a test covering this would be good. Will add that to the next revision.

Reply:
added tests in the next revision.